tests/test.py

```diff
@@ -1091,47 +1119,68 @@ def tearDown(self):
         if not self.TEST_DATA_LOG:
             return
-        with open('{}/{}_zone'.format(KEYS_FOLDER.name, zones[0][ZONE_ID]), 'rb') as f:
-            zone_private = f.read()
-        with open('{}/{}'.format(KEYS_FOLDER.name, acra_key_name), 'rb') as f:
-            private_key = f.read()
-        with open('{}/{}.pub'.format(KEYS_FOLDER.name, acra_key_name), 'rb') as f:
-            public_key = f.read()
-        logging.debug("test log: {}".format(json.dumps(
-            {
-                'master_key': get_master_key(),
-                'key_name': acra_key_name,
-                'private_key': b64encode(private_key).decode('ascii'),
-                'public_key': b64encode(public_key).decode('ascii'),
-                'data': b64encode(data).decode('ascii'),
-                'expected': b64encode(expected).decode('ascii'),
-                'zone_private': b64encode(zone_private).decode('ascii'),
-                'zone_public': zones[0][ZONE_PUBLIC_KEY],
-                'zone_id': zones[0][ZONE_ID],
-                'poison_record': b64encode(get_poison_record()).decode('ascii'),
-            }
-        )))
+
+        def key_name():
+            if storage_client_id:
+                return 'client storage, id={}'.format(storage_client_id)
+            elif zone_id:
+                return 'zone storage, id={}'.format(zone_id)
+            elif poison_key:
+                return 'poison record key'
+            else:
+                return 'unknown'
+
+        log_entry = {
+            'master_keys': get_master_keys(),
+            'key_name': key_name(),
+            'data': b64encode(data).decode('ascii'),
+            'expected': b64encode(expected).decode('ascii'),
+        }
+
+        if storage_client_id:
+            public_key = read_storage_public_key(storage_client_id, KEYS_FOLDER.name)
+            private_key = read_storage_private_key(KEYS_FOLDER.name, storage_client_id)
+            log_entry['public_key'] = b64encode(public_key).decode('ascii')
+            log_entry['private_key'] = b64encode(private_key).decode('ascii')
+
+        if zone_id:
+            public_key = read_zone_public_key(storage_client_id, KEYS_FOLDER.name)
+            private_key = read_zone_private_key(KEYS_FOLDER.name, storage_client_id)
+            log_entry['zone_public'] = b64encode(public_key).decode('ascii')
+            log_entry['zone_private'] = b64encode(private_key).decode('ascii')
+            log_entry['zone_id'] = zone_id
+
+        if poison_key:
+            public_key = read_poison_public_key(KEYS_FOLDER.name)
+            private_key = read_poison_private_key(KEYS_FOLDER.name)
+            log_entry['public_key'] = b64encode(public_key).decode('ascii')
+            log_entry['private_key'] = b64encode(private_key).decode('ascii')
+            log_entry['poison_record'] = b64encode(get_poison_record()).decode('ascii')
+
+        logging.debug("test log: {}".format(json.dumps(log_entry)))
```

Comment:
just interesting. something broke when you found this function and it needed refactoring or you had a need to see some log and found place where it produced?)

Reply:
Yeah, this function used to read key files directly from filesystem which does not work with v2. It's call sites relied on `acra_key_name` argument being treated as a file name too (e.g., to get access to the poison record key). So I've refactored it a bit to handle all those cases.

I've looked at its output, but it was not needed for debugging this PR.